std/math/emulated/field_mul.go

```diff
@@ -376,55 +435,11 @@ func (f *Field[T]) mulPreCond(a, b *Element[T]) (nextOverflow uint, err error) {
 	nbResLimbs := nbMultiplicationResLimbs(len(a.Limbs), len(b.Limbs))
 	nbLimbsOverflow := uint(1)
 	if nbResLimbs > 0 {
-		nbLimbsOverflow = uint(bits.Len(uint(2*nbResLimbs - 1)))
```

Comment:
can you explain this change ?

Reply:
Previously, we only had reduced multiplication and thus all the elements had same length `NbLimbs`. However, now this assumption doesn't always hold -- the quotient can be on less limbs in equality assertion, so it is more efficient to use the exact number of limbs.

Reply:
Oh, deleted previous comment. It was a bug previously.

When reducing the multiplication results always, then we could assume that every elements had always `nbLimbs` number of limbs. Then when computing the multiplications the result would have had `2 * nbLimbs - 1` limbs. This is important when we compute the overflow as for the carries computation we need to ensure we do not overflow. And the added overflow is `ceil(log2(2 * nbLimbs-1))`.

However, `nbMultiplicationResLimbs` already computes `len(a.Limbs) + len(b.Limbs) - 1` (and handling edge cases when len(a.Limbs) = 0 and len(b.Limbs)=1). In case `len(a.Limbs) = len(b.Limbs) = nbLimbs` then `nbResLimbs = 2*nbLimbs-1` already and we computed the overflow for the expected result of length `2*(2*nbLimbs-1)-1`, which is an overestimation.